src/main/java/org/jenkins/plugins/lockableresources/nodes/NodesMirror.java

```diff
@@ -0,0 +1,87 @@
+
+
+package org.jenkins.plugins.lockableresources;
+
+
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.Extension;
+import hudson.model.Node;
+import hudson.slaves.ComputerListener;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import jenkins.model.Jenkins;
+import jenkins.util.SystemProperties;
+
+//-----------------------------------------------------------------------------
+/** Mirror Jenkins nodes to lockable-resources
+*/
+@Extension
+public class NodesMirror extends ComputerListener {
+
+  private static final Logger LOG = Logger.getLogger(NodesMirror.class.getName());
+
+  //---------------------------------------------------------------------------
+  private static boolean isNodeMirrorEnabled() {
+    return SystemProperties.getBoolean("org.jenkins.plugins.lockableresources.ENABLE_NODE_MIRROR");
```

Comment:
Even if this is only used once (twice if you count the test as well) I'd suggest to move this into a `Constants.java` file or something